### PR TITLE
Michel/add godoc documentation

### PR DESCRIPTION
**Pull request by mlevieux**
Added godoc comments to exported entities

### CODE REVIEW

File: internal/crypto/crypto.go [hunk 1]
```diff
@@ -6,6 +6,9 @@ import (
 	"golang.org/x/crypto/curve25519"
 )
 
+// NewKeyPair returns a pair of cryptographic keys that can later
+// be used for signature and verification, along with an error,
```
Comment (blastrock) [inline internal/crypto/crypto.go:10]: Encryption!

File: config.go [hunk 2]
```diff
@@ -6,8 +6,11 @@ import (
 	"github.com/TankerHQ/identity-go/v3/internal/app"
 )
 
+// Config wraps information about an app
 type Config struct {
+	// AppID is the ID of the app corresponding to this config
 	AppID     string
+	// AppSecret is the app private signature key used to sign identities
```
Comment (blastrock) [inline config.go:13]: Not sure we should go into such details. From the point of view of the user, the app secret is opaque, they get it from the dashboard and that's all.

File: internal/crypto/crypto.go (hunk 1 above)
Comment (blastrock) [reply internal/crypto/crypto.go:10]: Sorry, I meant only encryption, or "encryption and decryption" if you want to be pedantic ^^

File: config.go (hunk 2 above)
Comment (blastrock) [reply config.go:13]: They can get these details by reading the spec, but we don't want anyone not prepared to do cryptography themselves. We don't expose cryptographic details in the API and we don't want someone to think "oh, this is a key, let me use that somewhere else in my app" and possibly write something very insecure.
You can just say something like "AppSecret is a secret used to create identities"

Comment (mlevieux) [reply config.go:13]: Okay thanks for the details! I'll update that :D

**Comment by mlevieux** [comment]
I think we're okay with this one, can I merge? (into feature/v3.2.0)

File: identity.go [hunk 3]
```diff
@@ -71,6 +74,8 @@ func CreateProvisional(config Config, target string, value string) (*string, err
 	return base64_json.Encode(provisional)
 }
 
+// GetPublicIdentity returns the public identity associated with the
+// identity it is sent
```
Comment (maximerety) [inline identity.go:78]:
```suggestion
// provided identity
```

File: identity.go [hunk 4]
```diff
@@ -116,6 +121,8 @@ func GetPublicIdentity(b64Identity string) (*string, error) {
 	return base64_json.Encode(publicIdentity)
 }
 
+// UpgradeIdentity upgrades the identity provided to it if needed and returns
```
Comment (maximerety) [inline identity.go:124]:
```suggestion
// UpgradeIdentity upgrades the provided identity if needed and returns
```

File: internal/app/app_secret.go [hunk 5]
```diff
@@ -14,6 +16,8 @@ const (
 	appCreationNature = 1
 )
 
+// GetAppId returns the app ID from the appSecret it is provided with.
```
Comment (maximerety) [inline internal/app/app_secret.go:19]:
```suggestion
// GetAppId returns the app ID from the provided appSecret.
```

File: internal/base64_json/base64_json.go [hunk 6]
```diff
@@ -29,6 +29,12 @@ func keySort(keys []string) {
 	})
 }
 
+// Encode returns a pointer to the base64 representation of the result of
+// marshalling v. If an error occurs in the process, it is returned.
+// Under the hood, Encode transforms v into a *orderedmap.OrderedMap so
+// the result will always wrap a key-sorted JSON representation to this
+// package. If v's underlying type is already *orderedmap.OrderedMap, v's
+// wrapped value will be used as is.
```
Comment (maximerety) [inline internal/base64_json/base64_json.go:37]:
```suggestion
// Encode returns a pointer to the base64 representation of the result of
// marshalling v in JSON. If an error occurs in the process, it is returned.
// Under the hood, Encode transforms v into a *orderedmap.OrderedMap so
// the result will always wrap a key-sorted JSON representation.
// If v's underlying type is already *orderedmap.OrderedMap, v's
// wrapped value will be used as is.
```

**Review by maximerety — CHANGES_REQUESTED** [review]

File: internal/base64_json/base64_json.go (hunk 6 above)
Comment (mlevieux) [reply internal/base64_json/base64_json.go:37]: `to this package` is here to remind that the key order is not a lexicographical one, but a custom one, internal to the package `base64_json`

Thoughts?

Comment (maximerety) [reply internal/base64_json/base64_json.go:37]: Maybe `the result will always wrap JSON representation in which keys are sorted in a specific order defined by this package.` or something like that. I did not understand the previous wording 😅

Comment (mlevieux) [reply internal/base64_json/base64_json.go]: I went for your suggestion. Figured as this package is internal, there's no need for such precision. Everything should have been updated.

**Review by maximerety — APPROVED** [review]

**Review by blastrock — APPROVED** [review]

**Pull request merged**